Fix various blocking haskell packages

mpscholten opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Fixes various haskell packages that are currently blocking a merge of haskell-updates by jailbreaking.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mpscholten committed:
haskellPackages.blucontrol: jailbreak

mpscholten committed:
haskellPackages.haskell-ci: jailbreak

mpscholten committed:
haskell.packages.ghc96.ghc-exactprint: use 1.7.1.0

dalpd commented:
Hi! `patat` should get a release soon with the updated bounds from https://github.com/jaspervdj/patat/pull/183. Though if you need to merge the other PR before that happens please go ahead, and I can remove the `jailbreak` in the next cycle.

maralorn commented:
Thank you for doing this @mpscholten. As one of the upstream maintainers of the affected packages I would very much appreciate a quick bug report about the bound. (Well, okay, I now know the problem with taskwarrior.^^)

mpscholten commented:
@dalpd thanks for mentioning, I'll drop the jailbreak from this branch then 👍 

@maralorn sorry, will open an issue next time 👍  I thought you might have seen my comment on the other PR already where I tagged everyone who maintains a broken build.

maralorn commented:
Oh, don’t worry. This is not so much about me. But generally our preferred style is to link an upstream issue in the comments.

sorki commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -2736,6 +2736,18 @@ self: super: {
+  taskwarrior = doJailbreak super.taskwarrior;
+
+  # jailbreak to allow network >= 3.2
+  zre = doJailbreak super.zre;
```
I've just uploaded a fix for this. I didn't notice it in the parent PR (sorry and thanks!). It should get fixed tomorrow so no need for jailbreak (but I don't mind reverting it myself later).

mpscholten replied:
thanks 👍 I'll drop the commit from this PR then

mpscholten commented:
I've gone ahead and added upstream issue links.

Also I dropped the taskwarrior fix, likely it's best to fix it directly in the upstream package (as you're the maintainer) and publish a new hackage release.

raehik commented:
Thanks for the upstream issue. binrep-1.1.0 fixes the GHC 9.8 build and should pass the next time the Hackage pin gets bumped.

raehik commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -2736,6 +2737,12 @@ self: super: {
+  blucontrol = doJailbreak super.blucontrol;
+
+  # jailbreak to allow rerefined >=0.7.0, https://github.com/raehik/binrep/issues/12
+  binrep = doJailbreak super.binrep;
```
should be good to remove next time Hackage pin is updated (binrep-1.1.0 uploaded 2025-03-11T12:11:30Z)

mpscholten replied:
Thanks, great, I'll drop the jailbreak from this PR 👍

mpscholten commented:
@maralorn @sternenseemann is this good to go, or are further changes needed?
